core/src/main/java/com/statful/client/core/sender/BufferedMetricsSender.java

```diff
@@ -316,4 +323,38 @@ private String buildAggregatedUri(final Aggregation aggregation, final Aggregati
 
         return buffersAsList;
     }
+
+    private void scheduleTask(final Runnable runnable) {
+        boolean wasScheduled = false;
+        if (executorService.getQueue().size() < configuration.getMaxWorkerTasksQueueSize()) {
```

Comment:
Well, I also considered that when developing this code but `ScheduledThreadPoolExecutor` is a `Futures` queue our abstraction used `Runables`, so I decided only to check the queue length to limit the queue size. Yeah... it sucks :/